### PR TITLE
moving future_mask to cuda for document attenion

### PR DESCRIPTION
**Pull request by KUNAL1612**
**Patch Description**
Moved the future mask to CUDA so that all operations for document attention also take place on CUDA. Refer to issue #285 for context. The speedup offered by this is marginal but over multiple runs may cumulate.

To test, I set attn doc seperator to a random number to trigger the code to enter the branch, and timed it using the methods used in #220. Observed that over 436 calls to the function on CPU it takes 0.42 seconds vs 0.09 seconds on GPU.

### CODE REVIEW

**Review by punitkoura — APPROVED** [review]

File: metaseq/models/transformer.py [hunk 1]
```diff
@@ -724,7 +724,7 @@ def buffered_future_mask(self, tensor, input_tokens=None):
         if need_to_make_new_mask:
             self._future_mask = torch.triu(
                 utils.fill_with_neg_inf(torch.zeros([max_seq_len, max_seq_len])), 1
-            )
+            ).to(tensor)
```
Comment (ngoyal2707) [inline metaseq/models/transformer.py:727]: instead of creating the tensor on cpu and moving to cuda,

you could just create on cuda like:
```
self._future_mask = torch.triu(
        utils.fill_with_neg_inf(torch.zeros([max_seq_len, max_seq_len], device=tensor.device)), 1
)
```

Comment (KUNAL1612) [reply metaseq/models/transformer.py]: updated to address this
